src/EnergyPlus/api/datatransfer.cc

```diff
@@ -60,7 +61,7 @@
 #include <EnergyPlus/OutputProcessor.hh>
 #include <EnergyPlus/PluginManager.hh>
 
-char * listAllAPIDataCSV() {
+char * listAllAPIDataCSV(EnergyPlusState) {
```

Comment:
Added `state` as an argument to all the API functions to future proof them.  Right now most of the functions don't end up using it, but they will once we stop getting everything off the global program state.